internal/events/event_consumer_test.go

```diff
@@ -78,40 +69,148 @@ func Test_EventConsumer_Consume_SendDLQ(t *testing.T) {
-		Return()
+		On("LogAndReportErrors", mock.Anything, mock.Anything, handlingErr.Error()).Return()
 
 	consumerMock.
-		On("Topic").
-		Return("test.test_topic").
-		On("ReadMessage", ctx).
-		Return(msg, nil).
-		On("Handlers").
-		Return([]EventHandler{&FailEventHandler{}})
+		On("Topic").Return("test.test_topic").
+		On("ReadMessage", ctx).Return(msg, nil).
+		On("Handlers").Return([]EventHandler{failedEventHandlerMock})
+
+	failedEventHandlerMock.
+		On("Handle", ctx, msg).Return(handlingErr).
+		On("CanHandleMessage", ctx, msg).Return(true).
+		On("Name").Return("FailedEventHandler")
 
 	producerMock.
-		On("WriteMessages", ctx, mock.Anything).
-		Return(nil)
+		On("WriteMessages", ctx, mock.Anything).Return(nil)
 
 	getEntries := log.DefaultLogger.StartTest(log.WarnLevel)
 
 	ec.Consume(ctx)
 
 	entries := getEntries()
-	assert.Equal(t, "Waiting 2s before retrying reading new messages", entries[0].Message)
+	assert.Equal(t, "Waiting 2s before retrying handling message with key key-1", entries[0].Message)
 	assert.Equal(t, "Max backoff reached for topic test.test_topic.", entries[1].Message)
 	assert.Equal(t, "Sending message with key key-1 to DLQ for topic test.test_topic", entries[2].Message) // backoffManager.ResetBackoff() was called
 
 	consumerMock.AssertExpectations(t)
 	crashTrackerMock.AssertExpectations(t)
-	producerMock.AssertExpectations(t)
+}
+
+func Test_EventConsumer_Consume_HandleFailedOnly(t *testing.T) {
+	// setup mocks
+	consumerMock := &MockConsumer{}
+	crashTrackerMock := &crashtracker.MockCrashTrackerClient{}
+	producerMock := NewMockProducer(t)
+	failedEventHandlerMock := NewMockEventHandler(t)
+	successfulEventHandlerMock := NewMockEventHandler(t)
+
+	handlingErr := errors.New("handling message for topic test.test_topic")
+	msg := &Message{Key: "key-1", Topic: "test.test_topic"}
+
+	ec := NewEventConsumer(consumerMock, producerMock, crashTrackerMock)
+
+	// Kill the consumer after 3 seconds.
+	ctx, cancel := context.WithDeadline(context.Background(), time.Now().Add(time.Second*3))
+	defer cancel()
+
+	crashTrackerMock.
+		On("LogAndReportErrors", mock.Anything, mock.Anything, handlingErr.Error()).
+		Return()
+
+	// In the 3 seconds runtime, we're expecting call `ReadMessage` once.
+	consumerMock.
+		On("ReadMessage", ctx).Return(msg, nil).Once().
+		On("Topic").Return("test.test_topic").
+		On("Handlers").Return([]EventHandler{failedEventHandlerMock, successfulEventHandlerMock})
+
+	// In the 3 seconds runtime, we're expecting to call `Handle` twice for `FailedEventHandler`.
+	// 1 call triggered by the original message and one backoff.
+	failedEventHandlerMock.
+		On("Handle", ctx, msg).Return(handlingErr).Twice().
+		On("CanHandleMessage", ctx, msg).Return(true).
+		On("Name").Return("FailedEventHandler")
+
+	// In the 3 seconds runtime, we're expecting to call `Handle` once for `SuccessfulEventHandler`.
+	//  This is triggered by the original message. The backoff shouldn't trigger this handler.
```

Comment:
nit:
```suggestion
	// This is triggered by the original message. The backoff shouldn't trigger this handler.
```

Reply:
Addressed in db76b256de5ccb95eb9ec5ecfb37bbb0675fd044